webapp/src/app.tsx

```diff
@@ -264,6 +266,23 @@ export class ProjectView
         this.shouldTryDecompile = false;
     }
 
+    openSerial(isSim: boolean) {
+        if (pxt.appTarget.serial && pxt.appTarget.serial.useEditor) {
+            let mainEditorPkg = pkg.mainEditorPkg()
+            if (mainEditorPkg) {
+                if (!mainEditorPkg.lookupFile("this/" + pxt.SERIAL_EDITOR_FILE)) {
+                    mainEditorPkg.setFile(pxt.SERIAL_EDITOR_FILE, "serial\n")
```

Comment:
this is not enough, pxt.json has to be updated as well.

Reply:
Post-beta, per discussion with Peli

Reply:
See https://github.com/Microsoft/pxt/issues/3114